src/build/__init__.py

```diff
@@ -59,6 +59,16 @@ class TypoWarning(Warning):
     """
 
 
+def _validate_source_directory(srcdir):
+    # type: (str) -> None
+    if not os.path.isdir(srcdir):
+        raise BuildException('Source {} is not a directory'.format(srcdir))
+    pyproject_toml = os.path.join(srcdir, "pyproject.toml")
+    setup_py = os.path.join(srcdir, "setup.py")
```

Comment:
You are not running pre-commit. Please use single quotes here :stuck_out_tongue:

Reply:
Thanks. Is the right way to run pre-commit to use the `tox -e fix` command? As I said, that's failing for me :-( Is there a contributing guide that explains how to set up for development, run tests, etc? I wasn't able to find one.

I tried running it by hand, which seems to have worked. So I've pushed a fix.

Reply:
It should, I don't know exactly what went wrong in your case :confused:. Perhaps @gaborbernat might have more insight?

Reply:
We should have a contributor guide, but right now I'm ETIME.

Reply:
No worries, I completely understand.

Reply:
> Thanks. Is the right way to run pre-commit to use the `tox -e fix` command? As I said, that's failing for me

Can you provide more exact details for this?

Reply:
I'll try to later, but basically it said I didn't have Visual C version 14 installed (I have VS 2019) when building the virtualenvs. But when I ran pre-commit outside of tox, it worked, and now `tox -e fix` works.

Reply:
Weird. Was it trying to build some c-extension package (such as black?)

Reply:
Yes. I can't recall which one, I'll let you know once I manage to reproduce the issue.